pkg/inject/sidecar_test.go

```diff
@@ -195,77 +196,145 @@ func TestSkipInjectSidecar(t *testing.T) {
-						corev1.Container{},
-					},
-				},
-			},
-		},
-	}
-
-	assert.False(t, Needed(dep))
-}
-
 func TestSidecarNeeded(t *testing.T) {
-	dep := dep(map[string]string{Annotation: "some-jaeger-instance"}, map[string]string{})
-	assert.True(t, Needed(dep))
-}
-
-func TestHasSidecarAlready(t *testing.T) {
-	dep := dep(map[string]string{Annotation: "TestHasSidecarAlready"}, map[string]string{})
-	assert.True(t, Needed(dep))
-	jaeger := v1.NewJaeger(types.NamespacedName{Name: "TestHasSidecarAlready"})
-	dep = Sidecar(jaeger, dep)
-	assert.False(t, Needed(dep))
-}
+	depWithAgent := dep(map[string]string{Annotation: "some-jaeger-instance"}, map[string]string{})
+	depWithAgent.Spec.Template.Spec.Containers = append(depWithAgent.Spec.Template.Spec.Containers, corev1.Container{
+		Name: "jaeger-agent",
+	})
 
-func TestSelectSingleJaegerPod(t *testing.T) {
-	dep := dep(map[string]string{Annotation: "true"}, map[string]string{})
-	jaegerPods := &v1.JaegerList{
-		Items: []v1.Jaeger{
-			v1.Jaeger{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "the-only-jaeger-instance-available",
-				},
-			},
+	tests := []struct {
+		dep    *appsv1.Deployment
+		ns     *corev1.Namespace
+		needed bool
+	}{
+		{
+			dep:    &appsv1.Deployment{},
+			ns:     &corev1.Namespace{},
+			needed: false,
+		},
+		{
+			dep:    dep(map[string]string{Annotation: "some-jaeger-instance"}, map[string]string{}),
+			ns:     ns(map[string]string{}),
+			needed: true,
+		},
+		{
+			dep:    dep(map[string]string{Annotation: "some-jaeger-instance"}, map[string]string{}),
+			ns:     ns(map[string]string{Annotation: "some-jaeger-instance"}),
+			needed: true,
+		},
+		{
+			dep:    dep(map[string]string{}, map[string]string{}),
+			ns:     ns(map[string]string{Annotation: "some-jaeger-instance"}),
+			needed: true,
+		},
+		{
+			dep:    depWithAgent,
+			ns:     ns(map[string]string{}),
+			needed: false,
 		},
 	}
-
-	jaeger := Select(dep, jaegerPods)
-	assert.NotNil(t, jaeger)
-	assert.Equal(t, "the-only-jaeger-instance-available", jaeger.Name)
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("dep:%s, ns: %s", test.dep.Annotations, test.ns.Annotations), func(t *testing.T) {
+			assert.Equal(t, test.needed, Needed(test.dep, test.ns))
+		})
+	}
 }
 
-func TestCannotSelectFromMultipleJaegerPods(t *testing.T) {
-	dep := dep(map[string]string{Annotation: "true"}, map[string]string{})
-	jaegerPods := &v1.JaegerList{
-		Items: []v1.Jaeger{
-			v1.Jaeger{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "the-first-jaeger-instance-available",
-				},
-			},
-			v1.Jaeger{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "the-second-jaeger-instance-available",
-				},
-			},
+func TestSelect(t *testing.T) {
+	jTest := v1.NewJaeger(types.NamespacedName{Name: "test"})
+	jProd := v1.NewJaeger(types.NamespacedName{Name: "prod"})
+	tests := []struct {
+		dep      *appsv1.Deployment
+		ns       *corev1.Namespace
+		jaegers  *v1.JaegerList
+		expected *v1.Jaeger
+		cap      string
```

Comment:
What does `cap` stands for?

Reply:
it's a caption. It's passed to `test.Run(cap, ..`. It helps to identify the failing test - if there is any.

Reply:
Got it. I was confused by the name, even though I got the purpose.